R/pkg.R

```diff
@@ -184,14 +184,25 @@ getPackageRecords <- function(pkgNames,
                               lib.loc = NULL,
                               missing.package = error_not_installed,
                               check.lockfile = FALSE,
-                              fallback.ok = FALSE)
+                              fallback.ok = FALSE,
+                              .visited.packages = new.env(parent = emptyenv()))
 {
   project <- getProjectDir(project)
   local.repos <- get_opts("local.repos", project = project)
 
   # screen out empty package names that might have snuck in
   pkgNames <- setdiff(pkgNames, "")
 
+  # Prior recursive steps may have already computed this package record and
+  # its recursive dependencies. Avoid constructing this package record.
+  priorPkgRecords <- c()
+  for (pkgName in pkgNames) {
+    if (exists(pkgName, envir = .visited.packages)) {
+      append(priorPkgRecords, get(pkgName, envir = .visited.packages))
```

Comment:
```suggestion
      priorPkgRecords <- append(priorPkgRecords, get(pkgName, envir = .visited.packages))
```

As is, this statement won't update `priorPkgRecords` -- is that intentional?